unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/BaseAvatar.cs

```diff
@@ -51,10 +54,21 @@ public async UniTask FadeOut(MeshRenderer targetRenderer, bool playParticles, Ca
             if (avatarRevealerContainer == null) 
                 return;
             
-            cancellationToken.ThrowIfCancellationRequested();
-
+            disposeCts ??= new CancellationTokenSource();
+            CancellationToken linkedCt = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, disposeCts.Token).Token;
+            linkedCt.ThrowIfCancellationRequested();
+            
             avatarRevealer.AddTarget(targetRenderer);
-            await avatarRevealer.StartAvatarRevealAnimation(playParticles, cancellationToken);
+            await avatarRevealer.StartAvatarRevealAnimation(playParticles && !isInsideHideArea, linkedCt);
+            
+            disposeCts?.Dispose();
+            disposeCts = null;
+        }
+        
+        public void ToggleHideAvatarArea(bool entered)
```

Comment:
I would make the `BaseAvatar` oblivious to hide areas and such to avoid coupling (at least conceptually by now).